### ISSUE TITLE
`desaturate` must be a first-class function

### ISSUE THREAD

**Issue by dompl** [oddbird/accoutrement-color#11]
Hiya.

I may be going nuts here, but I'm keep on getting the following error :

`Error: [string] 'desaturate' must be a first-class function`

My colour palette : 

```scss
$colors: (
  'brand': #889696,
  'gray': #cacaca,
  'gray-light': 'gray' ('shade': 25%, 'desaturate': 15%),
  'text': 'gray',
  'link': 'brand',
  'link-hover': 'brand' ('desaturate': 15%),
);
```
them I call the colour in another variable:
```scss
$navigation_link_colour: color('gray-light');
```

Not sure what am I doing wrong here.

Thanks

**Comment by mirisuzanne** [comment]
It sounds like you've upgraded Sass 3.5 which means functions are now first-class, and will soon be namespaced per file. That means you can no longer pass in function names, and assume we know what function they refer to. In the future, this will need to be something like…

```scss
$colors: (
  'brand': #889696,
  'gray': #cacaca,
  // 'shade' will be safe in our next release, because we provide it internally…
  'gray-light': 'gray' ('shade': 25%, get-function('desaturate'): 15%),
  'text': 'gray',
  'link': 'brand',
  'link-hover': 'brand' (get-function('desaturate'): 15%),
);
```

That's pretty ugly, but could be simplified to…

```scss
$desaturate: get-function('desaturate');
$colors: (
  'brand': #889696,
  'gray': #cacaca,
  'gray-light': 'gray' ('shade': 25%, $desaturate: 15%),
  'text': 'gray',
  'link': 'brand',
  'link-hover': 'brand' ($desaturate: 15%),
);
```

We'll keep looking for ways to improve this syntax, as the Sass module system is rolled out.

**Comment by mirisuzanne** [comment]
Improved fix in 51dccd4 allows you to capture functions in a shared map that we can use for lookup:

```scss
$functions: (
  'darken': get-function('darken'),
  'lighten': get-function('lighten'),
);
```

That way, you shouldn't have to make any changes to your existing color maps.
